mesonbuild/modules/unstable_external_project.py

```diff
@@ -0,0 +1,251 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os, subprocess, shlex
+
+from . import ExtensionModule, ModuleReturnValue
+from .. import mlog, build
+from ..mesonlib import (MesonException, Popen_safe, MachineChoice,
+                       get_variable_regex, do_replacement)
+from ..interpreterbase import InterpreterObject, InterpreterException, FeatureNew
+from ..interpreterbase import stringArgs, permittedKwargs
+from ..interpreter import DependencyHolder, InstallDir
+from ..compilers.compilers import cflags_mapping, cexe_mapping
+from ..dependencies.base import InternalDependency, PkgConfigDependency
+
+class ExternalProject(InterpreterObject):
+    def __init__(self, interpreter, subdir, project_version, subproject, environment, build_machine, host_machine,
+                 configure_command, configure_options, cross_configure_options, env, verbose):
+        InterpreterObject.__init__(self)
+        self.methods.update({'dependency': self.dependency_method,
+                             })
+
+        self.interpreter = interpreter
+        self.subdir = subdir
+        self.project_version = project_version
+        self.subproject = subproject
+        self.env = environment
+        self.build_machine = build_machine
+        self.host_machine = host_machine
+        self.configure_command = configure_command
+        self.configure_options = configure_options
+        self.cross_configure_options = cross_configure_options
+        self.verbose = verbose
+        self.user_env = env
+
+        self.name = os.path.basename(self.subdir)
+        self.src_dir = os.path.join(self.env.get_source_dir(), self.subdir)
+        self.build_dir = os.path.join(self.env.get_build_dir(), self.subdir, 'build')
+        self.install_dir = os.path.join(self.env.get_build_dir(), self.subdir, 'dist')
+        self.prefix = self.env.coredata.get_builtin_option('prefix')
+        self.libdir = self.env.coredata.get_builtin_option('libdir')
+        self.includedir = self.env.coredata.get_builtin_option('includedir')
+
+        # On Windows if the prefix is "c:/foo" and DESTDIR is "c:/bar", `make`
+        # will install files into "c:/bar/c:/foo" which is an invalid path.
+        # Work around that issue by removing the drive from prefix.
+        self.prefix = os.path.splitdrive(self.prefix)[1]
+
+        # self.prefix is an absolute path, so we cannot use it in
+        # os.path.join(something, self.prefix).
+        self.rel_prefix = self.prefix[1:]
```

Comment:
Is this also the case when `self.prefix` is `c:Program Files`?

Reply:
It's valid without `/` ? Damned Windows path :( Any idea how to make it relative in a portable way then?

Reply:
There is [this](https://stackoverflow.com/questions/23955968/path-with-no-slash-after-drive-letter-and-colon-what-does-it-point-to), so it probably isn't what it is intended but it sort of works.

`self.prefix = self.prefix.relative_to(self.prefix.drive + self.prefix.root)` is actually a protable way to solve this, since `self.prefix.root` will evaluate to an empty string in this case :P

Reply:
Ok, you mark a point for justifying moving to pathlib :D

And drive+root is anchor as far as I understand.

Reply:
True... Somehow I missed anchor.

Reply:
Updated to use pathlib, let me know if it's correct.